### PR TITLE
add nightly testing for minibude

### PR DESCRIPTION
**Pull request by stonea**
This PR adds nightly testing for the miniBUDE benchmark implemented by @xianghao-wang and @milthorpe.

This PR is essentially a copy of #24073, but rather than copying miniBude and its input data works by using an  executable `SKIPIF` file to download the benchmark and dataset from Github and then applies a patch that:

- eliminate warnings/errors in newer versions of Chapel
- make sure that the test can run in our suite in both correctness and performance modes

### CODE REVIEW

File: test/gpu/native/studies/minibude/SKIPIF [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+# Skip if not doing performance testing
+if [ -z "$CHPL_TEST_PERF" ]; then
+  echo "True"
+  exit
+fi
+
+CWD=$(cd $(dirname ${BASH_SOURCE[0]}) ; pwd)
+MINIBUDE_URL=${MINIBUDE_URL:-git@github.com:xianghao-wang/miniBUDE.git}
+MINIBUDE_BRANCH=${MINIBUDE_BRANCH:-benchmark}
```
Comment (e-kayrakli) [inline test/gpu/native/studies/minibude/SKIPIF:11]: @milthorpe mentioned [here](https://github.com/chapel-lang/chapel/pull/24073#issuecomment-1960671222) that the current version is in a different repo. Could you check whether the patch and the setup here works for that before merging?

Comment (stonea) [reply test/gpu/native/studies/minibude/SKIPIF]: Good point. I've updated things to use <https://github.com/milthorpe/miniBUDE/tree/v2/src/chapel>.

The code is a bit different so I've had to modify the patch file somewhat.

File: test/gpu/native/studies/minibude/README [hunk 2]
```diff
@@ -0,0 +1,13 @@
+This directory is for our nightly performance testing of miniBudert.
```
Comment (e-kayrakli) [inline test/gpu/native/studies/minibude/README:1]: Typo in minibude

Comment (e-kayrakli) [reply test/gpu/native/studies/minibude/README:1]: Could you also drop the link to the miniBUDE repo here?

File: test/gpu/native/studies/minibude/SKIPIF [hunk 3]
```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env bash
+
+# Skip if not doing performance testing
+if [ -z "$CHPL_TEST_PERF" ]; then
+  echo "True"
+  exit
+fi
```
Comment (e-kayrakli) [inline test/gpu/native/studies/minibude/SKIPIF]: reminder: we'll drop this part

**Review by e-kayrakli — APPROVED** [review]

**Pull request merged**